src/Shared/NativeMethodsShared.cs

```diff
@@ -918,25 +918,56 @@ internal static void KillTree(int processIdToKill)
         internal static int GetParentProcessId(int processId)
         {
             int ParentID = 0;
-            SafeProcessHandle hProcess = OpenProcess(eDesiredAccess.PROCESS_QUERY_INFORMATION, false, processId);
-
-            if (!hProcess.IsInvalid)
+            if (IsUnixLike)
             {
+                string line = null;
+
                 try
                 {
-                    // UNDONE: NtQueryInformationProcess will fail if we are not elevated and other process is. Advice is to change to use ToolHelp32 API's
-                    // For now just return zero and worst case we will not kill some children.
-                    PROCESS_BASIC_INFORMATION pbi = new PROCESS_BASIC_INFORMATION();
-                    int pSize = 0;
+                    // /proc/<processID>/stat returns a bunch of space separated fields. Get that string
+                    using (var r = FileUtilities.OpenRead("/proc/" + processId + "/stat"))
+                    {
+                        line = r.ReadLine();
+                    }
+                }
+                catch // Ignore errors since the process may have terminated
```

Comment:
I don't think it's possible at this time. I have no idea what kind of exception I may get. I may clarify this after some experimentation.